app/controllers/problems_controller.rb

```diff
@@ -0,0 +1,74 @@
+class ProblemsController < ApplicationController
+  before_action :set_problem, only: %I[show edit update destroy]
+  before_action :authenticate_user!, only: %I[new edit create destroy]
+
+  # GET /problems
+  # GET /problems.json
+  def index
+    @problems = Problem.all
```

Comment:
Do not forget to add here pagination. In case you have 1 000 000 problems in database all of them wiil be loaded here into RAM